src/core/Object3D.js

```diff
@@ -579,7 +580,14 @@ Object.assign( THREE.Object3D.prototype, THREE.EventDispatcher.prototype, {
 
 		if ( this.name !== '' ) object.name = this.name;
 		if ( JSON.stringify( this.userData ) !== '{}' ) object.userData = this.userData;
-		if ( this.castShadow === true ) object.castShadow = true;
+
+		if ( this.castShadow === true ) {
+
+			object.castShadow = true;
+			object.castShadowSize = this.castShadowSide;
```

Comment:
Better like this actually:

```js
if ( this.castShadow === true ) object.castShadow = true;
if ( this.castShadowSide !== THREE.BackSide ) object.castShadowSize = this.castShadowSide;
```

The user may have disabled `castShadow` temporarily.